src/plugins/share/public/components/tabs/export/export_content.tsx

```diff
@@ -200,21 +200,23 @@ const ExportContentUi = ({ isDirty, objectType, aggregateReportTypes, intl }: Ex
   }, [absoluteUrl, isDirty, renderCopyURLButton]);
 
   const getReport = useCallback(() => {
-    if (!generateReportForPrinting && !generateReport && !downloadCSVLens) {
+    if (!generateReport && !downloadCSVLens) {
       throw new Error('Report cannot be run due to no generate report method registered');
     }
-    if (objectType === 'lens' && selectedRadio === 'lens_csv') {
+    if (objectType === 'lens' && selectedRadio === 'csv') {
       return downloadCSVLens!();
+    } else if (usePrintLayout && selectedRadio === 'pngV2') {
+      return generateReport!({ intl });
     }
```

Comment:
I think we can get away from the conditional here  if we modified the `ShareMenuItem` interface to only accept for instance  `generateReport` handler, if we have a situation where the appropriate handler is  registered for the accompanying objectType, there will be no need to handle the logic of determining the appropriate handler to invoke. What do you think?

Reply:
Maybe we can look into this but it wasn't passing in tests or working when testing locally.